OpenRA.Mods.RA/Widgets/Logic/ClassicProductionLogic.cs

```diff
@@ -147,6 +147,43 @@ public ClassicProductionLogic(Widget widget, OrderManager orderManager, World wo
 					}
 				}
 			};
+
+			// Hook up scroll up and down buttons on the palette
+			var scrollDown = widget.GetOrNull<ButtonWidget>("SCROLL_DOWN_BUTTON");
+
+			if (scrollDown != null)
+			{
+				scrollDown.OnClick = palette.ScrollDown;
+				scrollDown.IsVisible = () => palette.TotalIconCount > (palette.MaxIconRows * palette.Columns);
+			}
+
+			var scrollUp = widget.GetOrNull<ButtonWidget>("SCROLL_UP_BUTTON");
+
+			if (scrollUp != null)
+			{
+				scrollUp.OnClick = palette.ScrollUp;
+				scrollUp.IsVisible = () => palette.TotalIconCount > (palette.MaxIconRows * palette.Columns);
+			}
+
+			SetMaximumVisibleRows(palette);
+		}
+
+		private static void SetMaximumVisibleRows(ProductionPaletteWidget productionPalette)
```

Comment:
We leave the explicit `private` out by convention. https://github.com/OpenRA/OpenRA/wiki/Coding-Standard